lib/src/validator/strict_dependencies.dart

```diff
@@ -18,6 +18,9 @@ import 'package:stack_trace/stack_trace.dart';
 
 /// Validates that Dart source files only import declared dependencies.
 class StrictDependenciesValidator extends Validator {
+  final AnalysisSessionManager analysisSessionManager =
```

Comment:
should this be a global? what are the costs to having a separate instance for different use cases?

Reply:
The cost is not very high.
Pub uses only parsed units, so there is not much to share.